src/services/cart.js

```diff
@@ -35,7 +35,17 @@ export default function Cart(oldCart) {
       items[key] = Object.assign({key: key}, item);
       items[key].quantity = 0;
     }
-    items[key].quantity += 1;
+    items[key].quantity += quantity;
+    this.updateTotals();
+  };
+
+  this.remove = (item) => {
+    const items = this.items;
+    const key = item.url;
+    items[key].quantity -= 1;
```

Comment:
`remove()` can only be called if the item is in the cart. In other words, the code here shouldn't check for it, right?

Reply:
> remove() can only be called if the item is in the cart

Why would it be so?
remove is a public method on a class. It knows nothing about what context it might be used in, especially it has no guarantees about content of items object.

Reply:
Fair point. This is all gone now anyway.